examples/v1beta1/pipelineruns/alpha/pipeline-emitting-results.yaml

```diff
@@ -20,21 +20,36 @@ spec:
       - name: task2
         taskSpec:
           results:
-            - name: array-results
-              type: array
-              description: The array results
+            - name: object-results
+              type: object
+              description: The object results
+              properties:
+                foo: {
+                  type: string
+                }
+                hello: {
+                  type: string
+                }
           steps:
             - name: write-array
               image: bash:latest
               script: |
                 #!/usr/bin/env bash
-                echo -n "[\"4\",\"5\",\"6\"]" | tee $(results.array-results.path)
+                echo -n "{\"foo\":\"bar\",\"hello\":\"world\"}" | tee $(results.object-results.path)
     results:
-      - name: echo-indexing-array-results
+      - name: array-results
+        type: array
+        description: whole array
+        value: $(tasks.task1.results.array-results[*])
+      - name: array-indexing-results
```

Comment:
Maybe example like:
``` yaml
- name: array-results
        type: array
        description: whole array
        value: ["$(tasks.task2.results.object-results.foo)"]
```
and
``` yaml
- name: object-results
        type: object
        description: whole object
        value:
           key1: $(tasks.task2.results.object-results.foo)
           key2: $(tasks.task2.results.object-results.bar)
```
would also be valuable?

Reply:
sure

Reply:
Just realized that these are not supported before, now they are supported! Thanks for catching these cases! @ywluogg